### ISSUE TITLE
Support for value format validation

### ISSUE THREAD

**Issue by od-cyera** [jlandersen/vscode-kafka#229]
Hi, 
Does this extension currently support validation support for message body?
If not, It could be nice to have json format validation (eg: if you forgot to remove a trailing `,`):
![image](https://user-images.githubusercontent.com/79370724/180181620-dd299f7e-5533-4315-b406-4e95750854d8.png)
Currently, the message is sent with bad format.

Also, might be nice to support custom [json-schema](https://json-schema.org/) validations.
![image](https://user-images.githubusercontent.com/79370724/180182135-be19bb9f-71a0-43e7-a9df-76b054c9e2d2.png)

Would love to hear what you think.
And thanks for the great extension :)

**Comment by angelozerr** [comment]
Indeed it should be nice, but I fear that it will not an easy task. The first thing to do is to study if we can use https://github.com/microsoft/vscode-json-languageservice to manage partial JSON validation, hover, completion, etc.

**Comment by od-cyera** [comment]
Yep, I think we can start with the first suggestion and see how it goes. 
Ill update here when i find time to work on this. Thanks :)

**Comment by od-cyera** [comment]
Hey,
I finally got to work on this a little, I have a working POC using `vscode-json-languageservice` in `KafkaFileDiagnostics`.
Do you think its fine to parse the json section with the json language service on every `KafkaFileDiagnostics.doDiagnostics`?

Feels like a bad design to me. but not sure whats a better fitting solution looks like.
Maybe we can think about creating another, more specific `JSONProducerBlock` that caches the parsed `JSONDocument`.

Also seems like the way to add diagnostics in an embeded document is a little hacky, from what i can tell, I'm supposed to clone the document, and replace all the characters in it with " " except for the embeded part. That way the diagnostics align on the original document. Not sure if you have any experience with this sort of stuff, but would love to know if you have an alternative solution.

**Comment by angelozerr** [comment]
Thanks so muuch foe studying this super nice  feature.

I think we should follow the same idea than vscode html language server which consumes html and CSS vscode language service since html can embedds CSS.

I suggest that you try to mimic the language mode used in vscode. See https://github.com/microsoft/vscode/blob/main/extensions/html-language-features/server/src/modes/cssMode.ts which support CSS features not only diagnostics but completion etc that we should support too for json

In another words I rhink we should have à JSonDocument per Block and bot à single JsonDocument
